src/controllers/ContinuedIllnessController.ts

```diff
@@ -0,0 +1,77 @@
+import Joi from '@hapi/joi';
+import { SessionMiddleware } from 'ch-node-session-handler';
+import { controller } from 'inversify-express-utils';
+import { BaseController } from './BaseController';
+import { FormValidator } from './validators/FormValidator';
+
+import { AuthMiddleware } from 'app/middleware/AuthMiddleware';
+import { IllnessReasonFeatureMiddleware } from 'app/middleware/IllnessReasonFeatureMiddleware';
+import { loggerInstance } from 'app/middleware/Logger';
+import { Appeal } from 'app/models/Appeal';
+import { Illness } from 'app/models/Illness';
+import { YesNo } from 'app/models/fields/YesNo';
+import { createSchema } from 'app/models/fields/YesNo.schema';
+import { CONTINUED_ILLNESS_PAGE_URI } from 'app/utils/Paths';
+import { Navigation } from 'app/utils/navigation/navigation';
+
+const template = 'continued-illness';
+
+const navigation : Navigation = {
+    previous(): string {
+        return '';
+    },
+    next(): string {
+        return '';
+    },
+    actions: (_: boolean) => {
+        return {
+            continue:'action=continue'
+        };
+    }
+};
+
+@controller(CONTINUED_ILLNESS_PAGE_URI, IllnessReasonFeatureMiddleware, SessionMiddleware, AuthMiddleware)
+export class ContinuedIllnessController extends BaseController<any> {
+
+    constructor() {
+        const errorMessage = 'You must tell us if this is a continued illness';
+        const schema: Joi.AnySchema = Joi.object({
+            continuedIllness: createSchema(errorMessage)
+        }).unknown(true);
+
+        super(
+            template,
+            navigation,
+            new FormValidator(schema)
+        );
+    }
+
+    protected prepareViewModelFromAppeal(appeal: Appeal): any {
+        const illnessStartDateRaw = appeal.reasons.illness?.illnessStart;
+
+        if (!illnessStartDateRaw) {
+            throw new Error('Illness Start Date expected but not found');
+        } else {
+            const illnessStart = illnessStartDateRaw.toLocaleDateString('en-GB', {
+                year: 'numeric', month: 'long', day: 'numeric'
+            });
+            const hint: string = `You told us the illness started on ${illnessStart}`;
+            return { hint };
+        }
+    }
+
+    protected prepareSessionModelPriorSave(appeal: Appeal, value: Illness): Appeal {
```

Comment:
Value argument seems to have wrong type. Is should be generic type named `FORM` expected by `BaseController<FORM>`. As I mentioned earlier it should corrections to data structure of the form handled by POST handler.

Reply:
Again I can't see any instances of this in the codebase, and it doesn't compile unless I change the controller definition to:
`export class ContinuedIllnessController<FORM> extends BaseController<FORM> {`

No other top-level controllers in the codebase are generic in this way.

![Screenshot 2020-07-29 at 10 44 58](https://user-images.githubusercontent.com/43062389/88784978-8d9f3480-d188-11ea-9c10-0976d847ed26.png)

Reply:
Maybe others kept ignoring it when I wasn't looking :( `OtherReasonController` has it, although type isn't 100% correct cause it has attachments in it despite attachment wasn't in `request.body`. You can define form type locally in the controller class which is way better than ANY.

Reply:
Discussion continued on Slack. Added more specific Interface definition to controller and now using this type in definition. Marking as resolved.